src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js

```diff
@@ -7,62 +7,78 @@ import { removeHostsFromGroup } from '../utils/api';
 import componentTypes from '@data-driven-forms/react-form-renderer/component-types';
 import { Text } from '@patternfly/react-core';
 
-const schema = (groupName, hosts) => ({
-  fields: [
-    {
-      component: componentTypes.PLAIN_TEXT,
-      name: 'warning-message',
-      label:
-        hosts.length === 1 ? (
-          <Text>
-            <strong>{hosts[0].display_name}</strong> will no longer be part of{' '}
-            <strong>{groupName}</strong> and its configuration will be impacted.
-          </Text>
-        ) : (
-          <Text>
-            <strong>{hosts.length}</strong> systems will no longer be part of{' '}
-            <strong>{groupName}</strong> and their configuration will be
-            impacted.
-          </Text>
-        ),
-    },
-  ],
-});
+const schema = (hosts) => {
+  const hostsInGroup = hosts.filter(({ groups }) => groups.length > 0); // selection can contain ungroupped hosts
+  const groupName = hostsInGroup[0].groups[0].name;
+
+  return {
+    fields: [
+      {
+        component: componentTypes.PLAIN_TEXT,
+        name: 'warning-message',
+        label:
+          hostsInGroup.length === 1 ? (
+            <Text>
+              <strong>{hostsInGroup[0].display_name}</strong> will no longer be
+              part of <strong>{groupName}</strong> and its configuration will be
+              impacted.
+            </Text>
+          ) : (
+            <Text>
+              <strong>{hostsInGroup.length}</strong> systems will no longer be
+              part of <strong>{groupName}</strong> and their configuration will
+              be impacted.
+            </Text>
+          ),
+      },
+    ],
+  };
+};
+
+const statusMessages = (hosts) => {
+  const hostsInGroup = hosts.filter(({ groups }) => groups.length > 0);
+  const groupName = hostsInGroup[0].groups[0].name;
+
+  return hostsInGroup.length === 1
```

Comment:
You could also put all the checks inside the string, using a Template Literal with the ternary operator, and by that remove some code duplication :)

Something like this:
```
const numOfSystems = hostsInGroup.length;
return {
onSuccess: {
title: `${numOfSystems} system${numOfSystems===1 ? " was" : "s were"} removed from ${groupName}`,
},
onError: {
title: `Failed to remove ${numOfSystems} system${numOfSystems===1 ? "" : "s"} from ${groupName}`,
},
};
```

Reply:
Yeah this is the variant that used to be here. But I decided to rework it for easier code readability (even though it's not ideally following DRY).